packages/synapse-react-client/src/synapse-queries/dataaccess/useDataAccessSubmission.ts

```diff
@@ -114,8 +114,18 @@ export function useSubmitDataAccessRequest(
     SynapseClientError,
     { request: CreateSubmissionRequest; accessRequirementId: string }
   >(
-    ({ request }): Promise<ACTSubmissionStatus> =>
-      SynapseClient.submitDataAccessRequest(request, accessToken!),
+    async ({ request, accessRequirementId }): Promise<ACTSubmissionStatus> => {
+      // The subjectId or subjectType may not be known, so pick an arbitrary one off of the AR (SWC-6490)
+      if (request.subjectId == null || request.subjectType == null) {
+        const { subjects } = await SynapseClient.getSubjects(
+          accessToken,
+          accessRequirementId,
+        )
+        request.subjectId = subjects[0].id
+        request.subjectType = subjects[0].type
+      }
```

Comment:
We were doing this in SWC but didn't port this behavior over to synapse-react-client:

https://github.com/Sage-Bionetworks/SynapseWebClient/blob/e7910e64d26e19a7ae68064822c26d7fee37453f/src/main/java/org/sagebionetworks/web/server/servlet/DataAccessClientImpl.java#L83